bootlib/bootlib.c

```diff
@@ -0,0 +1,80 @@
+//
+// Created by iliaz on 9/10/2024.
+//
+
+#include "bootlib.h"
+
+/**
+ * resets the boot counter.
+ * should be called after nominal operation is observed
+ */
+void reset_boot_counter()
+{
+    uint32_t variables=(uint32_t)(&__variables_start__);// "&" does not dereference a pointer. The value of __variables_start__ is put to the data pointer
+    uint32_t *data = (uint32_t *) variables;
+    uint32_t data_buffer[IFLASH_PAGE_SIZE/sizeof(uint32_t)];
```

Comment:
```suggestion
    static uint32_t data_buffer[IFLASH_PAGE_SIZE/sizeof(uint32_t)];
```

This is negotiable but I'd rather not fill the stack with arrays (and then can keep a single variable for the whole lib - just need to refill it with Fs every time)